controllers/vmware/vspherecluster_reconciler.go

```diff
@@ -112,7 +112,8 @@ func (r ClusterReconciler) Reconcile(ctx goctx.Context, req ctrl.Request) (_ ctr
 
 	// Handle deleted clusters
 	if !vsphereCluster.DeletionTimestamp.IsZero() {
-		return r.reconcileDelete(clusterContext), nil
+		r.reconcileDelete(clusterContext)
```

Comment:
No idea why the linter suddenly complains about these findings: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-vsphere/2069/pull-cluster-api-provider-vsphere-test-release-1-6/1682374202553798656